pkg/agent/plugin/svidstore/gcpsecretmanager/gcloud.go

```diff
@@ -0,0 +1,290 @@
+	}
+}
+
+func (p *SecretManagerPlugin) SetLogger(log hclog.Logger) {
+	p.log = log
+}
+
+// Configure configures the SecretMangerPlugin.
+func (p *SecretManagerPlugin) Configure(ctx context.Context, req *configv1.ConfigureRequest) (*configv1.ConfigureResponse, error) {
+	// Parse HCL config payload into config struct
+	config := &Configuration{}
+	if err := hcl.Decode(config, req.HclConfiguration); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to decode configuration: %v", err)
+	}
+
+	if len(config.UnusedKeys) != 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "unknown configurations detected: %s", strings.Join(config.UnusedKeys, ","))
+	}
+
+	client, err := p.hooks.newClient(ctx, config.ServiceAccountFile)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to create secretmanager client: %v", err)
+	}
+
+	// gcp secret manager does not allow ".", hash td as label
+	tdHash := sha1.Sum([]byte(req.CoreConfiguration.TrustDomain)) //nolint: gosec // We use sha1 to hash trust domain names in 128 bytes to avoid secret label restrictions
+
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	p.client = client
+	p.tdHash = hex.EncodeToString(tdHash[:])
+
+	return &configv1.ConfigureResponse{}, nil
+}
+
+// PutX509SVID puts the specified X509-SVID in the configured Google Cloud Secrets Manager
+func (p *SecretManagerPlugin) PutX509SVID(ctx context.Context, req *svidstorev1.PutX509SVIDRequest) (*svidstorev1.PutX509SVIDResponse, error) {
+	opt, err := optionsFromSecretData(req.Metadata)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get secret, if it does not exist, a secret is created
+	secret, err := getSecret(ctx, p.client, opt.secretName(), p.tdHash)
+	if err != nil {
+		return nil, err
+	}
+
+	// Secret not found, create it
+	if secret == nil {
+		secret, err = p.client.CreateSecret(ctx, &secretmanagerpb.CreateSecretRequest{
+			Parent:   opt.parent(),
+			SecretId: opt.name,
+			Secret: &secretmanagerpb.Secret{
+				Replication: &secretmanagerpb.Replication{
+					Replication: &secretmanagerpb.Replication_Automatic_{
+						Automatic: &secretmanagerpb.Replication_Automatic{},
+					},
+				},
+				Labels: map[string]string{
+					"spire-svid": p.tdHash,
+				},
+			},
+		})
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to create secret: %v", err)
+		}
+		p.log.With("secret_name", secret.Name).Debug("Secret created")
+
+		if opt.roleName != "" && opt.serviceAccount != "" {
+			// Create a policy without conditions and a single binding
+			resp, err := p.client.SetIamPolicy(ctx, &iam.SetIamPolicyRequest{
+				Resource: opt.secretName(),
+				Policy: &iam.Policy{
+					Bindings: []*iam.Binding{
+						{
+							Role:    opt.roleName,
+							Members: []string{opt.serviceAccount},
+						},
+					},
+				},
+			})
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to set IAM policy to secret: %v", err)
```

Comment:
```Thinking further on this .. what if someone changes the selectors on the entry,```
every time someone change a selector current "secret" will be removed before creating a new one.
It was done this way to avoid adding loginc to know when we must update a "secret" and when it is possible or not, so just create a new one with the same name after old secret was removed